consensus/istanbul/types_v2_test.go

```diff
@@ -183,3 +187,187 @@ func TestRoundChangeV2RLPEncoding(t *testing.T) {
 		t.Fatalf("RLP Encode/Decode mismatch. Got %v, expected %v", result.Request, original.Request)
 	}
 }
+
+// **** Big size prepreparev2 test
+
+func fillF(slice []byte, length int) {
+	for i := 0; i < length && i < len(slice); i++ {
+		slice[i] = 0xFF
+	}
+}
+
+func getF(length int) []byte {
+	f := make([]byte, length)
+	fillF(f, length)
+	return f
+}
+
+func bigF(length int) *big.Int {
+	even := new(big.Int).Lsh(common.Big1, uint(length))
+	return new(big.Int).Sub(even, common.Big1)
+}
+
+func writeExtra(header *types.Header, validators int) error {
+	extra := types.IstanbulExtra{
+		AddedValidators:           make([]common.Address, validators),
+		AddedValidatorsPublicKeys: make([]blscrypto.SerializedPublicKey, validators),
+		RemovedValidators:         big.NewInt(int64(validators)),
+		Seal:                      getF(64),
+		AggregatedSeal: types.IstanbulAggregatedSeal{
+			Bitmap:    bigF(validators),
+			Signature: getF(48),
+			Round:     common.Big32,
+		},
+		ParentAggregatedSeal: types.IstanbulAggregatedSeal{
+			Bitmap:    bigF(validators),
+			Signature: getF(48),
+			Round:     common.Big32,
+		},
+	}
+	for i := 0; i < validators; i++ {
+		extra.AddedValidators[i] = common.BytesToAddress(getF(20))
+		fillF(extra.AddedValidatorsPublicKeys[i][:], 96)
+	}
+	payload, err := rlp.EncodeToBytes(&extra)
+	if err != nil {
+		return err
+	}
+
+	if len(header.Extra) < types.IstanbulExtraVanity {
+		header.Extra = append(header.Extra, bytes.Repeat([]byte{0x00}, types.IstanbulExtraVanity-len(header.Extra))...)
+	}
+	header.Extra = append(header.Extra[:types.IstanbulExtraVanity], payload...)
+	return nil
+}
+
+func bigHeader(validators int) *types.Header {
+	header := &types.Header{
+		ParentHash:  common.BigToHash(common.Big257),
+		Coinbase:    common.BigToAddress(common.Big32),
+		Root:        common.BigToHash(common.Big1),
+		TxHash:      common.BigToHash(common.Big256),
+		ReceiptHash: common.BigToHash(common.Big3),
+		Bloom:       [types.BloomByteLength]byte{},
+		Number:      big.NewInt(37),
+		GasUsed:     123213,
+		Time:        100,
+	}
+	fillF(header.Bloom[:], types.BloomByteLength)
+	writeExtra(header, validators)
+	return header
+}
+
+func bigTxs(gasLimit int) []*types.Transaction {
+	curr := common.BytesToAddress(getF(20))
+	gateFeeRec := common.BytesToAddress(getF(20))
+	tx := types.NewTransaction(999, common.BytesToAddress(getF(20)),
+		big.NewInt(329274), 2942729, big.NewInt(294279),
+		&curr,
+		&gateFeeRec, big.NewInt(32027), getF(gasLimit/4))
```

Comment:
extract that 4 as a different variable, and add a comment that that's the cost of an empty byte

Reply:
Even better. Use the `TxDataZeroGas` constant in the `params/protocol_params.go` which is actually setting that scenario gas price